bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/internal/Ism8Handler.java

```diff
@@ -0,0 +1,183 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ism8.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.ism8.server.DataPointChangedEvent;
+import org.openhab.binding.ism8.server.IDataPoint;
+import org.openhab.binding.ism8.server.IDataPointChangeListener;
+import org.openhab.binding.ism8.server.Server;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Ism8Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hans-Reiner Hoffmann - Initial contribution
+ */
+@NonNullByDefault
+public class Ism8Handler extends BaseThingHandler implements IDataPointChangeListener {
+
+    private final Logger logger = LoggerFactory.getLogger(Ism8Handler.class);
+
+    private @Nullable Ism8Configuration config;
+    private @Nullable Server server;
+
+    public Ism8Handler(Thing thing) {
+        super(thing);
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Ism8: Handle command = {} {}", channelUID.getId(), command);
+        if (isLinked(channelUID)) {
+            Channel channel = null;
+            for (Channel ch : getThing().getChannels()) {
+                if (ch.getUID().getId().equals(channelUID.getId())) {
+                    channel = ch;
+                    break;
+                }
+            }
+
+            if (channel != null) {
+                if (channel.getConfiguration().containsKey("id") && channel.getConfiguration().containsKey("write")
+                        && channel.getConfiguration().get("write").toString().equalsIgnoreCase("true")) {
+                    int id = Integer.parseInt(channel.getConfiguration().get("id").toString());
+                    logger.info("Channel '{}' writting into ID '{}'", channel.getUID().getId(), id);
+                    updateState(channelUID, new QuantityType<>(command.toString()));
+
+                    IDataPoint dataPoint = null;
+                    for (IDataPoint dp : server.getDataPoints()) {
+                        try {
+                            if (dp.getId() == id) {
+                                dataPoint = dp;
+                                break;
+                            }
+                        } catch (Exception e) {
+                        }
```

Comment:
What are you expecting to catch here?

Reply:
Basically nothing, but Eclipse initially requested this in order to build it successfully. I could solve this by reviewing the Exception handling as previously were a lot of methods decorated by the ‘throws Exception’ statement. I removed them at all possible methods and only a few had to stay.